### PR TITLE
Fixes #33122 - Introduce skip pattern for ACL test

### PR DESCRIPTION
**Pull request by ezr-ondrej**
The ACL tests are testing all the routes, but if some test previous to
ACL tests add some routes, it fails as such dummy test routes do not
have proper perms. We need to skip such paths in tests as currently we
rely on order of test run.

### CODE REVIEW

File: test/unit/shared/access_permissions_test_base.rb [hunk 1]
```diff
@@ -5,15 +5,19 @@ module AccessPermissionsTestBase
   extend ActiveSupport::Concern
 
   module ClassMethods
-    def check_routes(app_routes, skipped_actions)
+    def should_skip_path?(path, skipped_actions = [], skip_patterns = [])
+      skipped_actions.include?(path) || skip_patterns.any? { |pattern| path =~ pattern }
+    end
+
+    def check_routes(app_routes, skipped_actions, skip_patterns: nil)
```
Comment (tbrisker) [inline test/unit/shared/access_permissions_test_base.rb]: Should this be 
```suggestion
    def check_routes(app_routes, skipped_actions, skip_patterns: [/^(api\/v2\/)?(dummy|fake)_/])
```

File: test/unit/shared/access_permissions_test_base.rb [hunk 2]
```diff
@@ -5,15 +5,19 @@ module AccessPermissionsTestBase
   extend ActiveSupport::Concern
 
   module ClassMethods
-    def check_routes(app_routes, skipped_actions)
+    def should_skip_path?(path, skipped_actions = [], skip_patterns = [])
```
Comment (tbrisker) [inline test/unit/shared/access_permissions_test_base.rb]: `skip_actions` and `skip_patterns` are always passed, no point in the defaults here (defaults are only used if you don't pass a parameter, if a param is passed it takes precedent even if it's `nil`)

**Review by tbrisker — CHANGES_REQUESTED** [review]
Needs a redmine issue (and perhaps some details in the commit message on why it's needed)

File: test/unit/shared/access_permissions_test_base.rb (hunk 2 above)
Comment (ezr-ondrej) [reply test/unit/shared/access_permissions_test_base.rb]: Yeah, I overengineered it a bit, as it has only one user as of now, no reason for defaults :)

File: test/unit/shared/access_permissions_test_base.rb (hunk 1 above)
Comment (ezr-ondrej) [reply test/unit/shared/access_permissions_test_base.rb]: Yeah, probably... I wanted to allow passing `nil`, but there is no reason for it.

**Comment by ezr-ondrej** [comment]
Sorry for not marking it as WIP, I've only extracted it from the puppet PR not to lose it.

I've fixed it and it's ready for re-review :)

**Review by tbrisker — APPROVED** [review]
thanks @ezr-ondrej !

**Pull request merged**